tar/tar.bzl

```diff
@@ -31,6 +31,31 @@ tar(
     mutate = mutate(strip_prefix = package_name()),
 )
 ```
+
+Compression is supported. Of course it takes additional time and resources.
+
+```starlark
+# Build this target to produce archive.tar.gz:
+tar(
+    name = "my_archive",
+    compress = "gzip",
+    ...
+)
+```
+
+Normally if your tar file is small, you can keep using the built-in gzip, but if your tar file is large, over 1GB, parallel `pigz` will greatly improve the performance.
+
+1. Add `pigz` in `MODULE.bazel`; see https://registry.bazel.build/modules/pigz
+2. Use the new tar compressor attribute in your BUILD file, eg:
+
+```starlark
+tar(
+    name = "my_tar_file",
+    compressor = "@pigz",
```

Comment:
Is this not toolchainized? Specifying the compressor per-target feels a bit off to me.

Reply:
It's not, there has been some discussion of where such a toolchain symbol would live, surely it's not here though.